pkg/util/log.go

```diff
@@ -1,117 +1,75 @@
-func InitLogger(level AllowedLevel) {
-	Logger = MustNewPrometheusLogger(level)
-}
+	logMessages = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "log_messages_total",
+		Help: "Total number of log messages.",
+	}, []string{"level"})
 
-// LogLevel supports registering a flag for the desired log level.
-type LogLevel struct {
-	AllowedLevel
-}
+	supportedLevels = []level.Value{
+		level.DebugValue(),
+		level.InfoValue(),
+		level.WarnValue(),
+		level.ErrorValue(),
+	}
+)
 
-// RegisterFlags adds the log level flag to the provided flagset.
-func (l *LogLevel) RegisterFlags(f *flag.FlagSet) {
-	l.Set("info")
-	f.Var(
-		&l.AllowedLevel,
-		"log.level",
-		"Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal]",
-	)
+func init() {
+	prometheus.MustRegister(logMessages)
 }
 
-// WithContext returns a Logger that has information about the current user in
-// its details.
-//
-// e.g.
-//   log := util.WithContext(ctx)
-//   log.Errorf("Could not chunk chunks: %v", err)
-func WithContext(ctx context.Context, l log.Logger) log.Logger {
-	// Weaveworks uses "orgs" and "orgID" to represent Cortex users,
-	// even though the code-base generally uses `userID` to refer to the same thing.
-	userID, err := user.ExtractOrgID(ctx)
+// InitLogger initialises the global gokit logger (util.Logger) and overrides the
+// default logger for the server.
+func InitLogger(cfg *server.Config) {
+	l, err := NewPrometheusLogger(cfg.LogLevel)
 	if err != nil {
-		return l
+		panic(err)
 	}
-	return WithUserID(userID, l)
-}
 
-// WithUserID returns a Logger that has information about the current user in
-// its details.
-func WithUserID(userID string, l log.Logger) log.Logger {
-	// See note in WithContext.
-	return log.With(l, "org_id", userID)
+	Logger = l
+	cfg.Log = logging.GoKit(l)
 }
 
 // PrometheusLogger exposes Prometheus counters for each of go-kit's log levels.
 type PrometheusLogger struct {
-	counterVec *prometheus.CounterVec
-	logger     log.Logger
+	logger log.Logger
 }
 
-var supportedLevels = []level.Value{level.DebugValue(), level.InfoValue(), level.WarnValue(), level.ErrorValue()}
-
-// NewPrometheusLogger creates a new instance of PrometheusLogger which exposes Prometheus counters for various log levels.
-// Contrarily to MustNewPrometheusLogger, it returns an error to the caller in case of issue.
-// Use NewPrometheusLogger if you want more control. Use MustNewPrometheusLogger if you want a less verbose logger creation.
-func NewPrometheusLogger(al AllowedLevel) (log.Logger, error) {
+// NewPrometheusLogger creates a new instance of PrometheusLogger which exposes
+// Prometheus counters for various log levels.
+func NewPrometheusLogger(l logging.Level) (log.Logger, error) {
 	// This code copy-pasted from prometheus/common/promlog.New()
-	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
-	l = al.Filter(l)
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
```

Comment:
Is the comment still correct?  ("This code copy-pasted from prometheus/common/promlog.New()")

Reply:
Not in any meaningful way, no.  Will remove.